### DISCUSSION TITLE
setting specific cuda id

### DISCUSSION THREAD

**Discussion by steveowk** [opendilab/LightZero#268] [Q&A] [answered]
Hello, 

Needed help with setting specific cuda ID.
Current config files seem to use cuda:0 by default.
Tx.

**Comment by puyuan1996** [comment, marked as answer]
One of the simplest ways to specify which CUDA device to use is by setting the environment variable using the following command:

```bash
export CUDA_VISIBLE_DEVICES=<cuda_id>
```

This command should be executed before starting your program to ensure that the desired CUDA device is used.

Alternatively, you can set the environment variable directly within the configuration file or script. For example, in Python, you can set the environment variable programmatically before running the rest of your code:

```python
import os
os.environ["CUDA_VISIBLE_DEVICES"] = "<cuda_id>"
```

This allows you to control the CUDA device within the code, making it easier to switch devices without altering the command line.

We plan to integrate the explicit passing of CUDA IDs directly in the LightZero configuration in the future, which will provide more flexibility and simplify the process further.

Thank you!
